e2e/hardhatTransactionFlow.spec.js

```diff
@@ -15,6 +18,40 @@ let account = null;
 const RAINBOW_WALLET_DOT_ETH = '0x7a3d05c70581bD345fe117c06e45f9669205384f';
 const TESTING_WALLET = '0x3Cb462CDC5F809aeD0558FBEe151eD5dC3D3f608';
 
+const ETH_ADDRESS = 'eth';
+const BAT_TOKEN_ADDRESS = '0x0d8775f648430679a709e98d2b0cb6250d2887ef';
+const CRYPTOKITTIES_ADDRESS = '0x06012c8cf97BEaD5deAe237070F9587f8E7A266d';
+
+const getOnchainBalance = async (address, tokenContractAddress) => {
+  const provider = new JsonRpcProvider(
```

Comment:
Can we do a singleton style function for this, since it's being reused here:  https://github.com/rainbow-me/rainbow/pull/2604/files#diff-6184d169993b8bec23fc1e6f3f721fc1a92d65f46c2f74f528a6975a4790129aL20
?

if it's already created return from memory it otherwise create it.